### PR TITLE
Fix `http.url` Zipkin span tag and add `http.protocol` and `http.serfmt`

### PR DESCRIPTION
**Pull request by trustin**
Motivation:

The Zipkin span tag `http.url` has to be a valid addressable URL that
starts with `http://` or `https://`. We currently use `Scheme.uriText()`
which may be surprising to users.

Modifications:

- Make sure `http.url` starts with `http://` or `https://` for HTTP(S)
  endpoints.
- Add `http.protocol` and `http.serfmt`.

Result:

Less surprising behavior.

### CODE REVIEW

File: zipkin/src/test/java/com/linecorp/armeria/client/tracing/HttpTracingClientTest.java [hunk 1]
```diff
@@ -108,12 +106,13 @@ public void shouldSubmitSpanWhenSampled() throws Exception {
...
-                "http.path", "/hello/armeria",
-                "http.status_code", "200",
-                "http.url", "none+h2c://foo.com/hello/armeria"));
+        assertThat(span.tags()).containsEntry("http.host", "foo.com")
+                               .containsEntry("http.method", "POST")
+                               .containsEntry("http.path", "/hello/armeria")
+                               .containsEntry("http.status_code", "200")
+                               .containsEntry("http.url", "http://foo.com/hello/armeria")
+                               .containsEntry("http.protocol", "h2c")
+                               .containsEntry("http.serfmt", "none");
```
Comment (adriancole) [inline zipkin/src/test/java/com/linecorp/armeria/client/tracing/HttpTracingClientTest.java]: as this creates index load, possibly add when not none?

Comment (trustin) [reply zipkin/src/test/java/com/linecorp/armeria/client/tracing/HttpTracingClientTest.java]: Updated.

File: zipkin/src/main/java/com/linecorp/armeria/internal/tracing/SpanTags.java [hunk 2]
```diff
@@ -73,6 +81,32 @@ public static void addTags(Span span, RequestLog log) {
         }
     }
 
+    private static String generateUrl(Scheme scheme, String authority, String path, @Nullable String query) {
+        final SessionProtocol sessionProtocol = scheme.sessionProtocol();
+        final String uriScheme;
+        if (SessionProtocol.httpValues().contains(sessionProtocol)) {
+            uriScheme = "http://";
+        } else if (SessionProtocol.httpsValues().contains(sessionProtocol)) {
+            uriScheme = "https://";
```
Comment (adriancole) [inline zipkin/src/main/java/com/linecorp/armeria/internal/tracing/SpanTags.java]: is this one a pain to test? I noticed it wasn't in the test case, but not terribly fussed

Comment (trustin) [reply zipkin/src/main/java/com/linecorp/armeria/internal/tracing/SpanTags.java]: I can add one for sure.

Comment (trustin) [reply zipkin/src/main/java/com/linecorp/armeria/internal/tracing/SpanTags.java]: Added `SpanTagsTest` :smile:

**Comment by trustin** [comment]
Thanks, reviewers :+1:

**Pull request merged**